extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelHelper.java

```diff
@@ -0,0 +1,271 @@
+package org.schabi.newpipe.extractor.services.youtube;
+
+import com.grack.nanojson.JsonObject;
+import com.grack.nanojson.JsonWriter;
+import org.schabi.newpipe.extractor.exceptions.ContentNotAvailableException;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.localization.ContentCountry;
+import org.schabi.newpipe.extractor.localization.Localization;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.defaultAlertsCheck;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getJsonPostResponse;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.prepareDesktopJsonBuilder;
+import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
+
+/**
+ * Shared functions for extracting YouTube channel pages and tabs.
+ */
+public final class YoutubeChannelHelper {
+    private YoutubeChannelHelper() {
+    }
+
+    /**
+     * Take a YouTube channel ID or URL path, resolve it if necessary and return a channel ID.
+     *
+     * @param idOrPath a YouTube channel ID or URL path
+     * @return a YouTube channel ID
+     * @throws IOException if a channel resolve request failed
+     * @throws ExtractionException if a channel resolve request response could not be parsed or is
+     * invalid
+     */
+    @Nonnull
+    public static String resolveChannelId(@Nonnull final String idOrPath)
+            throws ExtractionException, IOException {
+        final String[] channelId = idOrPath.split("/");
+
+        if (channelId[0].startsWith("UC")) {
+            return channelId[0];
+        }
+
+        // If the URL is not a /channel URL, we need to use the navigation/resolve_url endpoint of
+        // the InnerTube API to get the channel id.
+        // Otherwise, we couldn't get information about the channel associated with this URL, if
+        // there is one.
+        if (!channelId[0].equals("channel")) {
```

Comment:
~~I would change this by putting `if (channelId[0].equals("channel")) { return channelId[1]; }` at the top and therefore de-indenting what now is in the `if`.~~ Oh, I just saw that this `if` can passthrough. Maybe you can just extend this comment to explain that if the `browseId` extraction fails then the unresolved id will be returned.

Reply:
Also, I think there is an extra "such as" in the javadoc of this function